eg/motor.js

```diff
@@ -1,35 +1,33 @@
-var five = require("../lib/johnny-five.js"),
-  board, motor, led;
+const {Board, Motor} = require("../lib/johnny-five.js");
+const board = new Board();
 
-board = new five.Board();
-
-board.on("ready", function() {
+board.on("ready", () => {
   // Create a new `motor` hardware instance.
-  motor = new five.Motor({
+  const motor = new Motor({
     pin: 5
   });
 
   // Inject the `motor` hardware into
   // the Repl instance's context;
   // allows direct command line access
   board.repl.inject({
-    motor: motor
+    motor
   });
 
   // Motor Event API
 
   // "start" events fire when the motor is started.
-  motor.on("start", function() {
+  motor.on("start", () => {
     console.log("start", Date.now());
 
     // Demonstrate motor stop in 2 seconds
-    board.wait(2000, function() {
+    board.wait(2000, () => {
       motor.stop();
     });
```

Comment:
Could you also do `board.wait(2000, motor.stop)` or do you want to be specific about calling `motor.stop()` as a function?

Reply:
I like @HipsterBrown's suggestion here. It's nice to see that functions can be passed in a variable. We use anonymous function expressions as params all over the place. This is a gentle next step.

I remember first grokking this concept by seeing both used on setTimeout examples back in the day.

Reply:
I went through and did this, but I think I went too far. Whenever there is a parameter being passed to the callback like this:
````js
motor.on("stop", () => console.log("stop"));
````
It can get hard to read with a moderately complex callback, for example:
````js
motor.on("stop", () => console.log(`Motor stopped at ${Date.now()}));
````
or
````js
spdt.on("close", () => motor.stop().reverse(motor.speed()));
````

I propose that we use @HipsterBrown's pattern if there is no param passed to the callback, @rwaldron's pattern if there is a static value or single variable being passed to the callback, or just leave it as a block if there is something that makes it more difficult (chaining, templates, multiple params, etc).

Reply:
I've updated all these to the compromise I describe above. An example where all three are used is motor-directional.js

Reply:
> Could you also do board.wait(2000, motor.stop) or do you want to be specific about calling motor.stop() as a function?

Sure! That's even nicer :)

Reply:
So... I may have made a mistake in agreeing to @HipsterBrown's suggestion above :|

Node.js's EventEmitter binds the callback's `this` to the instance:

```js
const Emitter = require("events");

class A {
  write(pin, bit) {
    console.log(pin, bit);
  }
  brake() {
    console.log(this instanceof A === true);
    this.write(1, 0);
  }
}

const a = new A();
const e = new Emitter();

e.on("x", a.brake);
e.emit("x");
```

Results in: 

```
false
/Users/rwaldron/codez/javascript/wrong-this.js:9
    this.write(1, 0);
         ^

TypeError: this.write is not a function
```

Reply:
Unfortunately, there is no hard rule here... you'd have to run the code to ensure that it's not broken.

Reply:
I fixed the ones I could find... which might've just been one? http://johnny-five.io/examples/esc-bidirectional/

Reply:
I probably added a couple in motor and servo. I'll PR for motor and go patch servo right now before it lands.